src/docfx/docset/RepositoryProvider.cs

```diff
@@ -13,60 +13,26 @@ internal class RepositoryProvider
             = new ConcurrentDictionary<(FileOrigin origin, string dependencyName), Lazy<(string docset, Repository repository)>>();
 
         private readonly string _locale;
+        private readonly Func<RestoreGitMap> _restoreGitMap;
+        private readonly Func<Config> _config;
+        private readonly Lazy<(string fallbackDocsetPath, Repository fallbackRepo)> _fallbackDocset;
+
         private string _docsetPath;
```

Comment:
since this is mutable, it'll produce different result depending on when GetRepository is called.